### PR TITLE
Qt1-129 consume apprenticeship confirmation confirmed event

### PR DESCRIPTION
**Pull request by alireid**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAccounts.Jobs/EventHandlers/Simulator/SimulateEvents.cs [hunk 1]
```diff
@@ -38,5 +41,39 @@ internal class SimulateEvents
...
+                await endpoint.Publish(@event, executionContext, log);
+
+                return new AcceptedResult();
+            }
+            catch (Exception e)
+            {
+                return new BadRequestObjectResult(e);
+            }
+        }
+
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeAccounts.Jobs/EventHandlers/Simulator/SimulateEvents.cs:77]: These look like events taken from CMAD Jobs, most of which we don't consume. However, it's a security concern creating this Simulate endpoint (as it is a http endpoint, I think).  We spoke to someone about it sometime in the past and you you find a message publisher project for sending the message.

File: src/SFA.DAS.ApprenticeAccounts.Jobs.Tests/WhenApprenticeshipConfirmationConfirmed.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            ApprenticeshipConfirmationConfirmed sut,
+            ApprenticeshipConfirmationConfirmedEvent evt
+            )
+        {
+            await sut.Handle(evt, new TestableMessageHandlerContext());
+
+            api.Verify(m => m.SendApprenticeshipConfirmed(
+                evt.ApprenticeId,
+                It.IsAny<ApprenticeshipConfirmedCommand>()
+            ));
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeAccounts.Jobs.Tests/WhenApprenticeshipConfirmationConfirmed.cs:27]: I think we get more value if we explicitly check the values of the command being passed in. If we don't do this we can miss property mapping issues

**Review by Pauljgraham — COMMENTED** [review]
Some comments, worth discussing

File: src/SFA.DAS.ApprenticeAccounts.Jobs/EventHandlers/ApprenticeshipConfirmationEventHandlers/ApprenticeshipConfirmationConfirmed.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+using NServiceBus;
+using SFA.DAS.ApprenticeAccounts.Jobs.Api;
+using SFA.DAS.ApprenticeAccounts.Jobs.InternalMessages.Commands;
+using SFA.DAS.ApprenticeCommitments.Messages.Events;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeAccounts.Jobs.EventHandlers.LoginServiceEventHandlers
+{
+    public class ApprenticeshipConfirmationConfirmed : IHandleMessages<ApprenticeshipConfirmationConfirmedEvent>
+    {
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.ApprenticeAccounts.Jobs/EventHandlers/ApprenticeshipConfirmationEventHandlers/ApprenticeshipConfirmationConfirmed.cs:11]: Thats good @Pauljgraham, I'll make these changes with that in mind

File: src/SFA.DAS.ApprenticeAccounts.TestMessagePublisher/Program.cs [hunk 4]
```diff
@@ -28,6 +28,7 @@
 transport.AddRouting(routeSettings =>
 {
     routeSettings.RouteToEndpoint(typeof(LoginEmailAddressUpdated), queueName);
+    routeSettings.RouteToEndpoint(typeof(ApprenticeshipConfirmationConfirmed), queueName);
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.ApprenticeAccounts.TestMessagePublisher/Program.cs:31]: @Pauljgraham Gotcha, removed

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**